packages/api/lambdas/create-reconciliation-report.js

```diff
@@ -151,6 +146,99 @@ async function reconciliationReportForCollections() {
   };
 }
 
+/**
+ * Compare the granule holdings in CMR with Cumulus
+ *
+ * @param {string} collectionId - the collection which has the granules to be reconciled
+ * @returns {Promise<Object>} an object with the okGranulesInDb, okGranulesInCmr, onlyInCumulus,
+ * onlyInCmr
+ */
+async function reconciliationReportForGranules(collectionId) {
+  // compare granule holdings:
+  //   Get CMR granules list (by PROVIDER, short_name, version, sort_key: ['granule_ur'])
+  //   Get CUMULUS granules list (by collectionId order by granuleId)
+  //   Report granules only in CMR
+  //   Report granules only in CUMULUS
+  const { name, version } = deconstructCollectionId(collectionId);
+  const cmrGranulesIterator = new CMRSearchConceptQueue(
+    process.env.cmr_provider, process.env.cmr_client_id, 'granules',
+    { short_name: name, version: version, sort_key: ['granule_ur'] }, 'umm_json'
+  );
+
+  const dbGranulesIterator = new Granule().getGranulesForCollection(collectionId);
+
+  const okGranulesInDb = [];
+  const okGranulesInCmr = [];
+  const granulesOnlyInDb = [];
+  const granulesOnlyInCmr = [];
+
+  let [nextDbItem, nextCmrItem] = await Promise.all([dbGranulesIterator.peek(), cmrGranulesIterator.peek()]); // eslint-disable-line max-len
+
+  while (nextDbItem && nextCmrItem) {
+    const nextDbGranuleId = nextDbItem.granuleId;
+    const nextCmrGranuleId = nextCmrItem.umm.GranuleUR;
+
+    if (nextDbGranuleId < nextCmrGranuleId) {
```

Comment:
So just to make sure I have this correct, this method of comparing will only work if the granule IDs come back sorted. Are we positive that is always going to be the case?